pennylane/transforms/compile.py

```diff
@@ -149,6 +155,7 @@ def qfunc(x, y, z):
         else:
             # Expands out anything that is not a single operation (i.e., the templates)
             expanded_tape = tape.expand(stop_at=lambda obj: obj.name in all_ops)
+            expanded_tape = expanded_tape.expand(stop_at=lambda obj: obj.name != "Barrier")
```

Comment:
```suggestion
            stop_at = lambda obj: (obj.name in all_ops) and (obj.name != "Barrier)
            expanded_tape = tape.expand(stop_at=stop_at)
```

So the point of this is to eliminate `Barrier`'s when `only_visual=True`? Shouldn't the `remove_barrier` transform do that?